overlord/snapstate/snapstate.go

```diff
@@ -47,9 +47,13 @@ const (
 	// always be enforcing, even if the snap requests otherwise.
 	JailMode
 
-	// if we need flags for just SnapSetup it may be easier
-	// to start a new sequence from the other end with:
-	// 0x40000000 >> iota
+	// flags just for SnapSetup follow
+
+	// HasABA is a feature flag; doInstall sets it, and downstream handlers check for it to handle old SnapSetups
+	HasABA = 0x40000000 >> iota
```

Comment:
Let's please avoid this flag. It seems very mysterious, and we don't really need it since we always have the data at hand: before the task, we know because we were able to compute the flag in the first place. After the task, we are writing it down so that we can recover the original data on an undo. If we need a migration because the clean version of this logic is incompatible, let's please write the migration so we can have sane logic thereafter.